src/yunohost/app.py

```diff
@@ -89,27 +90,22 @@ def app_fetchlist(url=None, name=None):
 
     """
     # Create app path if not exists
-    try: os.listdir(repo_path)
-    except OSError: os.makedirs(repo_path)
+    if not os.path.exists(repo_path):
+        os.makedirs(repo_path)
 
     if url is None:
         url = 'https://app.yunohost.org/official.json'
         name = 'yunohost'
-    else:
-        if name is None:
-            raise MoulinetteError(errno.EINVAL,
-                                  m18n.n('custom_appslist_name_required'))
-
-    list_file = '%s/%s.json' % (repo_path, name)
-    if os.system('wget "%s" -O "%s.tmp"' % (url, list_file)) != 0:
-        os.remove('%s.tmp' % list_file)
-        raise MoulinetteError(errno.EBADR, m18n.n('appslist_retrieve_error'))
+    elif name is None:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('custom_appslist_name_required'))
 
-    # Rename fetched temp list
-    os.rename('%s.tmp' % list_file, list_file)
+    try:
+        urlretrieve(url, '%s/%s.json' % (repo_path, name))
+    except Exception as e:
+        raise MoulinetteError(errno.EBADR, m18n.n('appslist_retrieve_error'), error=str(e))
 
-    os.system("touch /etc/cron.d/yunohost-applist-%s" % name)
-    os.system("echo '00 00 * * * root yunohost app fetchlist -u %s -n %s > /dev/null 2>&1' >/etc/cron.d/yunohost-applist-%s" % (url, name, name))
+    open("/etc/cron.d/yunohost-applist-%s" % name, "w").write('00 00 * * * root yunohost app fetchlist -u %s -n %s > /dev/null 2>&1\n' % (url, name))
```

Comment:
This might be needed later when someone adds code after this line. Also, as we usually like to copy-paste code, this might spread elsewhere where it would have bad consequences.